src/utils/functionRuntimeUtils.ts

```diff
@@ -0,0 +1,84 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as opn from 'opn';
+import * as semver from 'semver';
+import * as vscode from 'vscode';
+import { extensionPrefix } from '../../src/ProjectSettings';
+import { DialogResponses } from '../DialogResponses';
+import { localize } from '../localize';
+import { cpUtils } from './cpUtils';
+
+export namespace functionRuntimeUtils {
+
+    const runtimePackage: string = 'azure-functions-core-tools';
+
+    export async function validateFunctionRuntime(outputChannel: vscode.OutputChannel): Promise<void> {
+        try {
+            const localVersion: string | null = await getLocalFunctionRuntimeVersion();
+            if (localVersion === null) {
+                return;
+            }
+            const newestVersion: string | null = await getNewestFunctionRuntimeVersion(semver.major(localVersion));
+            if (newestVersion === null) {
+                return;
+            }
+            if (semver.gt(newestVersion, localVersion)) {
+                await promptDocumentationAction(
+                    localize(
+                        'azFunc.outdatedFunctionRuntime',
+                        '[Azure Functions] Your installed version of the Azure Functions Core Tools ("{0}") does not match the latest ("{1}"). Please update for the best experience.',
+                        localVersion,
+                        newestVersion
+                    )
+                );
+            }
+        } catch (error) {
+            outputChannel.appendLine(`Error occurred when checking function runtime: ${error}`);
```

Comment:
It's okay to call it 'function runtime' in the code, but for anything displayed to the user, it should be called 'Azure Functions Core Tools' to be consistent with all the [docs](https://docs.microsoft.com/azure/azure-functions/functions-run-local)